docs/source/_notebooks/keras-text-classifiers.rst

```diff
@@ -0,0 +1,4124 @@
+
+
+.. parsed-literal::
+
+    [[   1 4825   14    9   87   21   24   38   87]]
+    [['<START>' 'hello' 'this' 'is' 'great' 'but' 'not' 'so' 'great']]
+
+
+Notice that this model does not require fixed length input. We do not
+need to pad this sample.
+
+.. code:: ipython3
+
+    binary_model.predict(review)
+
+
+
+
+.. parsed-literal::
+
+    array([[0.5912496]], dtype=float32)
+
+
+
+Neutral as expected.
+
+What makes the score go up?
+
+.. code:: ipython3
+
+    eli5.show_prediction(binary_model, review, tokens=review_t)
+
+
+
+
+.. raw:: html
+
+    
+        <style>
+        table.eli5-weights tr:hover {
+            filter: brightness(85%);
+        }
+    </style>
+    
+    
+    
+        
+    
+        
+    
+        
+    
+        
+    
+        
+    
+        
+    
+    
+        
+    
+        
+    
+        
+    
+        
+            
+    
+        
+    
+            
+    
+            
+    
+    
+        <p style="margin-bottom: 2.5em; margin-top:-0.5em;">
+            <span style="background-color: hsl(120, 100.00%, 62.45%); opacity: 0.98" title="0.047">&lt;START&gt;</span><span style="opacity: 0.80"> </span><span style="background-color: hsl(120, 100.00%, 60.69%); opacity: 1.00" title="0.050">hello</span><span style="opacity: 0.80"> </span><span style="background-color: hsl(120, 100.00%, 60.25%); opacity: 1.00" title="0.051">this</span><span style="opacity: 0.80"> </span><span style="background-color: hsl(120, 100.00%, 60.00%); opacity: 1.00" title="0.052">is</span><span style="opacity: 0.80"> </span><span style="background-color: hsl(120, 100.00%, 76.26%); opacity: 0.89" title="0.024">great</span><span style="opacity: 0.80"> </span><span style="background-color: hsl(120, 100.00%, 67.04%); opacity: 0.95" title="0.039">but</span><span style="opacity: 0.80"> </span><span style="background-color: hsl(120, 100.00%, 70.83%); opacity: 0.93" title="0.033">not</span><span style="opacity: 0.80"> </span><span style="background-color: hsl(120, 100.00%, 73.37%); opacity: 0.91" title="0.029">so</span><span style="opacity: 0.80"> </span><span style="background-color: hsl(120, 100.00%, 85.66%); opacity: 0.85" title="0.012">great</span>
+        </p>
+    
+    
+        
+    
+        
+    
+        
+    
+        
+    
+    
+        
+    
+        
+    
+        
+    
+        
+    
+        
+    
+        
+    
+    
+        
+    
+        
+    
+        
+    
+        
+    
+        
+    
+        
+    
+    
+    
+
+
+
+
+Now this is something. The words highlighted in green show what makes
+the score "go up", i.e. the "positive" words (check the next section to
+see how to show positive AND negative words with the ``relu`` argument).
+
+Hover over the highlighted words to see their "weight".
+
+Even though the explanation is bright green, the actual prediction is
+not very positive. Pass ``preserve_density=True`` to make the underlying
+formatter show colors "closer" to the values they represent.
```

Comment:
I'm not sure that using ``preserve_density=True`` here is the right thing to do, from the docs:

> If preserve_density is True, then color for longer fragments will be less intensive than for shorter fragments, so that "sum" of intensities will correspond to feature weight.
> If preserve_density is None, then it's value is taken from the corresponding attribute of doc_weighted_spans.

So it makes "is" brighter because it's shorter, I'm not sure whether it's a good thing or not, but if we want to leave that in, I'd prefer to have a more accurate description of what it does here.

Reply:
You are right. Sorry should've checked the docs. I think  it makes sense to remove this and just mention that the user should hover over text to see the actual value.

Reply:
Sounds good to me 👍